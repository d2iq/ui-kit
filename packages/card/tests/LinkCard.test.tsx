import React from "react";

import { LinkCard } from "../";
import { render } from "enzyme";
import toJSON from "enzyme-to-json";

describe("LinkCard", () => {
  it("renders default", () => {
    expect(
      toJSON(
        render(
          <LinkCard url="http://google.com" linkDescription="Google">
            Example Content
          </LinkCard>
        )
      )
    ).toMatchSnapshot();
  });
  it("renders with external link", () => {
    expect(
      toJSON(
        render(
          <LinkCard
            url="http://google.com"
            linkDescription="Google"
            openInNewTab={true}
          >
            Example Content
          </LinkCard>
        )
      )
    ).toMatchSnapshot();
  });
});
