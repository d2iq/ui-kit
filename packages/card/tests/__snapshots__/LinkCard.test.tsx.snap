// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkCard renders default 1`] = `
<div
  class="css-ix5gb9"
  data-cy="linkCard"
>
  <div
    class="css-1hc6qpp"
  >
    <a
      class="css-gsqxnx"
      href="http://google.com"
    >
      <span
        class="css-1uecx0y"
        data-cy="box"
      >
        Google
      </span>
    </a>
    Example Content
  </div>
</div>
`;

exports[`LinkCard renders with external link 1`] = `
<div
  class="css-ix5gb9"
  data-cy="linkCard"
>
  <div
    class="css-1hc6qpp"
  >
    <a
      class="css-gsqxnx"
      href="http://google.com"
      rel="noopener"
      target="_blank"
    >
      <span
        class="css-1uecx0y"
        data-cy="box"
      >
        Google
      </span>
    </a>
    Example Content
  </div>
</div>
`;
